### ISSUE TITLE
stubgen: Add typings of keys and values of dict (__members__, __entries)

### ISSUE THREAD

**Issue by jasminlapalme** [python/mypy#15888]
It would be useful to add typings of the keys and values of the `__members__` and `__entries` dictionary on class that represent `enum` in C.

For example, if we take the same example as #12717.

```c++
#include <cstddef>
#include <memory>
#include <pybind11/pybind11.h>

namespace py = pybind11;

enum class MyEnum { FOO, BAR };

PYBIND11_MODULE(native_enum_test, module) {
  pybind11::enum_<MyEnum>(module, "MyEnum", pybind11::arithmetic())
      .value("FOO", MyEnum::FOO)
      .value("BAR", MyEnum::BAR);
}
```

Compile via:

```
$ c++ -O3 -Wall -shared -std=c++17 -fPIC $(python3 -m pybind11 --includes) native_enum_test.cpp -o native_enum_test.so
```

Run the stub generator:

```
stubgen -p native_enum_test
```

the result is something like

```python
class MyEnum:
    __members__: ClassVar[dict] = ...  # read-only
    BAR: ClassVar[MyEnum] = ...
    FOO: ClassVar[MyEnum] = ...
    __entries: ClassVar[dict] = ...
...
```

When we import code with these kinds of stubs and use `mypy` to type checked the code with `strict` option, we have this warning.

```
error: Missing type parameters for generic type "dict"  [type-arg]
```

**Expected Behavior**

The result of `sturgeon` would be like this:

```python
class MyEnum:
    __members__: ClassVar[dict[str, MyEnum]] = ...  # read-only
    BAR: ClassVar[MyEnum] = ...
    FOO: ClassVar[MyEnum] = ...
    __entries: ClassVar[dict[str, tuple[MyEnum, None]]] = ... # not sure about the type of this one...
...
```

**Environment**

- Mypy version used: 1.4.1
- Mypy command-line flags: not relevant, this is about the stub generator `--show-column-numbers --no-error-summary --no-pretty --no-color-output`
- Mypy configuration options from `mypy.ini` (and other config files): strict = true
- Python version used: 3.9

**Comment by sobolevn** [comment]
Question about `__entries`: we usually don't include and annotate private members, why do you think that this one is needed?

**Comment by jasminlapalme** [comment]
In my case, I don't need `__entries`. But `mypy` generate an error with `strict` on `__entries` this is why I include it. I think that `stubgen` could ignore `__entries` and not expose this attribute.
